### PR TITLE
Add integration test case for invalid credentials

### PR DESCRIPTION
**Pull request by tedteng**
**How to categorize this PR?**

/area usability
/kind test
/platform azure

**What this PR does / why we need it**:
This PR tests https://github.com/gardener/terraformer/pull/93 and https://github.com/gardener/gardener/pull/3950 by adding an integration test case for invalid credentials.
It creates an infrastructure resource but uses invalid credentials and ensures that the ERR_INFRA_UNAUTHENTICATED is correctly detected from the termination message and that the infrastructure can still be deleted afterwards (as it's a no-op).

Before this PR, the destroy logic and testing was executed as part of a cleanup action, while the destroy logic should be part of the actual test and not just cleanup. Also, this could lead to resource leakage on AliCloud and on the test-machinery cluster, because if one cleanup action fails (e.g. the destroy logic) no further cleanup actions are executed.
Now, the destroy logic is executed as a deferred function to ensure that the cleanup actions are still run, even if the test itself fails.
**Which issue(s) this PR fixes**:
Fixes https://github.com/gardener/gardener-extension-provider-azure/issues/662

**Special notes for your reviewer**:

**Release note**:

```other operator
NONE
```
/invite @ialidzhikov

### CODE REVIEW

**Comment by tedteng** [comment]
/invite @ialidzhikov would you like to review this PR Thanks.

File: test/integration/infrastructure/infrastructure_test.go [hunk 1]
```diff
@@ -489,6 +491,94 @@ var _ = Describe("Infrastructure tests", func() {
...
+			var (
+				namespace *corev1.Namespace
+				cluster   *extensionsv1alpha1.Cluster
+				infra     *extensionsv1alpha1.Infrastructure
+			)
+
+			framework.AddCleanupAction(func() {
+				By("cleaning up namespace and cluster")
+				Expect(client.IgnoreNotFound(c.Delete(ctx, namespace))).To(Succeed())
+				Expect(client.IgnoreNotFound(c.Delete(ctx, cluster))).To(Succeed())
```
Comment (kon-angelo) [inline test/integration/infrastructure/infrastructure_test.go:511]: In the other tests, we have these in the defer part. Can I ask the reason to add these here instead ?

Comment (tedteng) [reply test/integration/infrastructure/infrastructure_test.go:511]: The code I refer to https://github.com/gardener/gardener-extension-provider-aws/pull/325/files#diff-159fa2b5244aa8582f3ab173f88010e046ae979b06ef94e88b0346b1d308dc50R272-R278.  It seems we don't want to leak any resources even if defer not called.

Comment (tedteng) [reply test/integration/infrastructure/infrastructure_test.go:511]: @kon-angelo

**Review by kon-angelo — APPROVED** [review]
/lgtm

**Pull request merged**